cpp/src/strings/convert/convert_datetime.cu

```diff
@@ -508,10 +524,34 @@ struct check_datetime_format {
       // reference: https://man7.org/linux/man-pages/man3/strptime.3.html
       bool result = false;
       switch (item.value) {
-        case 'Y': result = check_digits(ptr, item.length); break;
-        case 'y': result = check_digits(ptr, item.length); break;
-        case 'm': result = check_value(ptr, item.length, 1, 12); break;
-        case 'd': result = check_value(ptr, item.length, 1, 31); break;
+        case 'Y': {
+          if (auto value = str2int(ptr, item.length)) {
+            result             = true;
+            dateparts[TP_YEAR] = value.value();
+          }
+          break;
+        }
+        case 'y': {
+          if (auto value = str2int(ptr, item.length)) {
+            result             = true;
+            dateparts[TP_YEAR] = value.value() + 1900;
```

Comment:
As per the linked document (man3/strptime), the values [69,99] refer to 1969-1999 and the numbers [0,68] refer to 2000-2068. By the existing logic, 00 will be interpreted as 1900, which was not a leap year, whereas it should be interpreted as 2000 which *was* a leap year.

Reply:
Thanks Devavret. 
This would be a breaking change to the current API but I think is the correct approach and will fix it here.